src/test/groovy/ee/tuleva/onboarding/mandate/batch/MandateBatchIntegrationTest.java

```diff
@@ -147,7 +147,7 @@ void testMandateCreationBeforeRetirementAge() throws Exception {
             .build();
 
     when(withdrawalEligibilityService.getWithdrawalEligibility(any()))
-        .thenReturn(new WithdrawalEligibilityDto(false, 30, 55, false));
+        .thenReturn(new WithdrawalEligibilityDto(false, false, 30, 55, false));
```

Comment:
Any way to avoid boolean fields like this? `new WithdrawalEligibilityDto(false, false, 30, 55, false)` is not really understandable what they represent

Reply:
It's the unfortunate combination of record (no builders) + non-fleshed out abstraction. 
As we get more information on how other withdrawals are going to look like, we can refactor this DTO to something better. 
But even then we would probably write dead code we won't use to fill the "empty space" in the abstraction to make it complete.

For intellij, these boolean flags have the little name decorations which make it more readable ![Screenshot 2024-12-10 at 11 31 23](https://github.com/user-attachments/assets/033ea65d-f5ef-436d-b17f-f6bafd2230cf)

Reply:
Ah, nevermind. Builders can be used with record. But then problems with potentially nullable fields start, forgetting them etc.